lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3752,33 +3752,37 @@ static void diagnoseUnintendedOptionalBehavior(TypeChecker &TC, const Expr *E,
     }
 
     void visitInterpolatedStringLiteralExpr(InterpolatedStringLiteralExpr *E) {
-      // Warn about interpolated segments that contain optionals.
+      // Warn about interpolated segments that contain optionals or function.
       for (auto &segment : E->getSegments()) {
         // Allow explicit casts.
         if (auto paren = dyn_cast<ParenExpr>(segment))
           if (isa<ExplicitCastExpr>(paren->getSubExpr()))
             continue;
 
-        // Bail out if we don't have an optional.
-        if (!segment->getType()->getRValueType()->getOptionalObjectType())
+        bool isOptional = !!segment->getType()->getRValueType()->getOptionalObjectType();
```

Comment:
We don't typically do `!!` as an idiom to produce a `bool` (I know some projects like this). If you really want to make it explicit you can wrap the whole thing in `bool(...)`.

Reply:
I use `bool` instead of `!!`. https://github.com/apple/swift/pull/19234/commits/eff1b9eb63ba8c9a45e9587ba236133ebb788dc9